new podule plink2/het

RuthEberhardt opened the pull request:
<!--
# nf-core/modules pull request

Pull request fir new module plink2/het
This has been written to allow both --bfile and --pfile input options

Many thanks for contributing to nf-core/modules!

Please fill in the appropriate checklist below (delete whatever is not relevant).
These are the most common things requested on pull requests (PRs).

Remember that PRs should be made against the master branch.

Learn more about contributing: [CONTRIBUTING.md](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
-->

## PR checklist

Closes #7954 https://github.com/nf-core/modules/issues/7954 

- [ x] This comment contains a description of changes (with reason).
- [x ] If you've fixed a bug or added code that should be tested, add tests!
- [ x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ x] If necessary, include test data in your PR.
- [x ] Remove all TODO statements.
- [x ] Emit the `versions.yml` file.
- [x ] Follow the naming conventions.
- [x ] Follow the parameters requirements.
- [x ] Follow the input/output options guidelines.
- [x ] Add a resource `label`
- [ x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [x ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [x ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

jfy133 commented on modules/nf-core/plink2/het/main.nf:
```diff
@@ -0,0 +1,52 @@
+        'biocontainers/plink2:2.00a5.10--h4ac6f70_0' }"
+
+    input:
+    tuple val(meta), path(plink_file1), path(plink_file2), path(plink_file3)
```
Is there a specific reason why these are all in one tuple?

In the vast majority of cases it should be: one file == one channel [^1]

Also: I find `plint_file` `plink_file2` etc very informative, your descriptions in meta.yaml is very good though! maybe replace with `plink_genotypetable` , `plink_variantmap`, `plink_sampleinfo` or similar?

[^1]: https://nf-co.re/docs/guidelines/components/modules#required-path-channel-inputs

RuthEberhardt replied:
All 3 input files are required for plink or plink2 to work so I've left them in one tuple. I've changed the naming as in your suggestion - great suggestion thanks!
In commit https://github.com/nf-core/modules/pull/8081/commits/a12c4c50038b7e1193317117492539bbb2ad8acc

jfy133 commented on modules/nf-core/plink2/het/main.nf:
```diff
@@ -0,0 +1,52 @@
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    
```
```suggestion
```

RuthEberhardt replied:
Fixed in commit https://github.com/nf-core/modules/pull/8081/commits/a12c4c50038b7e1193317117492539bbb2ad8acc

jfy133 commented on modules/nf-core/plink2/het/meta.yml:
```diff
@@ -0,0 +1,55 @@
+name: "plink2_het"
+description: Calculate Inbreeding data in plink2
```
```suggestion
description: Calculate inbreeding data with plink2
```

RuthEberhardt replied:
fixed in commit https://github.com/nf-core/modules/pull/8081/commits/a12c4c50038b7e1193317117492539bbb2ad8acc

jfy133 commented on modules/nf-core/plink2/het/meta.yml:
```diff
@@ -0,0 +1,55 @@
+name: "plink2_het"
+description: Calculate Inbreeding data in plink2
+keywords:
+  - plink2
+  - inbreeding
+  - het
```
As an outsider: a few more would be nice here, what exactly does `het` correspond to for example

RuthEberhardt replied:
fixed in commit https://github.com/nf-core/modules/pull/8081/commits/a12c4c50038b7e1193317117492539bbb2ad8acc

jfy133 commented on modules/nf-core/plink2/het/meta.yml:
```diff
@@ -0,0 +1,55 @@
+         type: file
+         description: observed and expected homozygous/heterozygous genotype counts for each
+          sample and method-of-moments F coefficient estimates
+         pattern: "*.{het}"
```
```suggestion
         pattern: "*.het"
```

As it's a single file suffix

RuthEberhardt replied:
Fixed in commit https://github.com/nf-core/modules/pull/8081/commits/a12c4c50038b7e1193317117492539bbb2ad8acc
I've also made minor changes to the tests following review of a PR submitted by a colleague https://github.com/nf-core/modules/pull/8082

jfy133 reviewed (commented):
A few minor things, mostly for readability purposes

RuthEberhardt commented:
Thank you for your review James, really helpful suggestions for a nextflow novice. I've addressed all issues in commit https://github.com/nf-core/modules/commit/a12c4c50038b7e1193317117492539bbb2ad8acc so I hope this is OK. I;ve also made some changes to tests following a review a colleague has had for another plink module.
Please let me know if I need to do anything more

jfy133 commented on modules/nf-core/plink2/het/main.nf:
```diff
@@ -0,0 +1,51 @@
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        plink2: \$(plink2 --version)
```
Sorry, missed this the first time: this is not sufficient, you need to use `sed` or `cut` or similar to extract _just_ the numbers.

Currently the output of the command here is:

```
PLINK v2.00a5.12LM 64-bit Intel (25 Jun 2024)
```

But you need just

```
2.00a5.12LM
```

RuthEberhardt replied:
Fixed in commit https://github.com/nf-core/modules/pull/8081/commits/fb6416244927a7d70e6edacbb53f946b725dc66e

jfy133 commented on modules/nf-core/plink2/het/meta.yml:
```diff
@@ -0,0 +1,57 @@
+keywords:
+  - plink2
+  - inbreeding
+  - heterozygous genotypes
+  - homozygous genotypes
+  - F coefficient
```
Better! I think the `F` should be lower case

What about thins such as 'population genomics' ? What sort of discilpine is the module primarily used for

jfy133 reviewed (commented):
Almost there! Only blocker is the version :)

jfy133 commented:
Please send me a DM on slack @RuthEberhardt when ready for final review (I have too many github notifications atm so it might get lost)
